spec/src/main/java/io/a2a/spec/Message.java

```diff
@@ -129,6 +129,15 @@ public String getKind() {
         return kind;
     }
 
+    /**
+     * Creates a new Builder for constructing Message instances.
+     *
+     * @return a new Message.Builder instance
+     */
+    public static Builder builder() {
```

Comment:
I much prefer

```
Message m = Message.builder()
  [...]
  .build();
```

to 

```
Message m = new Message.Builder()
  [...]
  .build();
```

should we deprecate the constructor (or make it private)?

Reply:
As a bit of context, we used to have just the constructor variant to create these. The static `builder()` methods were added to some classes in order to clarify things for the MapStruct mappers.

I agree that `Message.builder()` is nicer, and if we have that everywhere, it makes things more consistent. 
@jmesnil feel free to open an issue. But I think for now we should just deprecate the public constructors and add the static builder methods. Then at a later date we can make the constructors private.

My reasoning is that we're already breaking a lot of stuff with this spec update. Then again maybe breaking more could be ok